google-cloud-spanner/src/main/java/com/google/cloud/spanner/MultiplexedSessionDatabaseClient.java

```diff
@@ -267,6 +299,59 @@ private void maybeMarkUnimplemented(Throwable t) {
     }
   }
 
+  private void maybeMarkUnimplementedForRW(SpannerException spannerException) {
+    if (spannerException.getErrorCode() == ErrorCode.UNIMPLEMENTED
+    // && spannerException.getCause().getMessage().contains("Transaction type read_write not
+    // supported with multiplexed sessions")
+    ) {
+      unimplementedForRW.set(true);
+    }
+  }
+
+  private void verifyBeginTransactionWithRWOnMultiplexedSession(String sessionName) {
```

Comment:
nit: rename this to the following to indicate that it is non-blocking

```suggestion
  private ApiFuture<Transaction> verifyBeginTransactionWithRWOnMultiplexedSessionAsync(String sessionName) {
```

Reply:
Done, I have updated the method name.
However, I have still kept the return type to be `void` since it is not being used.